lib/vcr.rb

```diff
@@ -329,18 +335,84 @@ def record_http_interaction(interaction)
     cassette.record_http_interaction(interaction)
   end
 
+  # @private
+  def link_context(from_thread, to_key)
+    @context[to_key] = get_context(from_thread)
+  end
+
+  # @private
+  def unlink_context(key)
+    @context.delete(key)
+  end
+
+  # @private
+  def fibers_available?
+    @fibers_available
+  end
+
 private
+  def current_context
+    get_context(Thread.current, Fiber.current)
+  end
+
+  def get_context(thread_key, fiber_key = nil)
+    context = @context[fiber_key] if fiber_key
+    context ||= @context[thread_key]
+    if context
+      context
+    else
+      @context[thread_key] = dup_context(@context[MainThread])
+    end
+  end
+
+  def context_value(name)
+    current_context[name]
+  end
+
+  def set_context_value(name, value)
+    current_context[name] = value
+  end
+
+  def dup_context(context)
+    {
+      :turned_off => context[:turned_off],
+      :ignore_cassettes => context[:ignore_cassettes].dup,
+      :cassettes => context[:cassettes].dup
+    }
+  end
 
   def ignore_cassettes?
-    @ignore_cassettes
+    context_value(:ignore_cassettes)
   end
 
   def cassettes
-    @cassettes ||= []
+    context_value(:cassettes)
+  end
+
+  def initialize_fibers
+    begin
+      require 'fiber'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.